### PR TITLE
added a prototype of RBF kernel

### PR DESCRIPTION
**Pull request by ASKabalan**

### CODE REVIEW

**Comment by ASKabalan** [comment]
@EiffL 

So I added three functions in file https://github.com/DifferentiableUniverseInitiative/JaxPM/blob/45-differentiable-painting-strategy-for-density-shells/jaxpm/spherical.py

They do NGP (first order not differentiable) bi-linear (second order) NGP RBF (still second order but with higher capacity of smoothing)

Painted maps look good

image

I also compare cl against limber approximation using this snipper to compute limber

```python
def compute_theory_cl():
    """Compute theoretical C_ell using Limber approximation (ℓ≥2)."""
    print("🧮 Computing theoretical C_ell with Limber approximation...")
    from jax.tree_util import register_pytree_node_class
    from jax_cosmo.redshift import redshift_distribution

    @register_pytree_node_class
    class tophat_z(redshift_distribution):
        def pz_fn(self, z):
            zmin, zmax = self.params
            return jnp.where((z >= zmin) & (z <= zmax), 1.0, 0.0)

    # Create cosmology (using same parameters as simulation)
    cosmo_th = jc.Planck15(Omega_c=omega_c, sigma8=sigma8)
    
    # Convert comoving distances to redshifts for shell limits
    a_min = jc.background.a_of_chi(cosmo_th, R_min).squeeze()
    a_max = jc.background.a_of_chi(cosmo_th, R_max).squeeze()
    zmin = float(jc.utils.a2z(a_min))
    zmax = float(jc.utils.a2z(a_max))
    
    print(f"   Distance shell: [{R_min:.1f}, {R_max:.1f}] Mpc/h")
    print(f"   Redshift range: z ∈ [{zmin:.3f}, {zmax:.3f}]")
    
    # Top-hat redshift window and unit bias (galaxy clustering, no RSD)
    nz = tophat_z(zmin, zmax, gals_per_arcmin2=1.0)
    bias = jc.bias.constant_linear_bias(1.0)
    probe = jc.probes.NumberCounts([nz], bias, has_rsd=False)
    
    # Setup integration grid and compute C_ell
    ell_theory = jnp.arange(0, LMAX + 1)
    cl_theory = jc.angular_cl.angular_cl(cosmo_th, ell_theory, [probe], nonlinear_fn=jc.power.linear).squeeze()
    
    return np.array(ell_theory[2:]), np.array(cl_theory[2:])
```

Here is the result

image

I tested here https://github.com/DifferentiableUniverseInitiative/JaxPM/blob/45-differentiable-painting-strategy-for-density-shells/tests/test_spherical_painting_methods.py

Not the ranges are quite large you might say 

https://github.com/DifferentiableUniverseInitiative/JaxPM/blob/45-differentiable-painting-strategy-for-density-shells/tests/test_spherical_painting_methods.py#L262

However at lmax the accuracy is pretty good and this is a relatively small nside (256)

I am adding a gradient test to finalize but I would appreciate your feedback

**Comment by ASKabalan** [comment]
@EiffL 

Added differentiation tests here

https://github.com/DifferentiableUniverseInitiative/JaxPM/blob/62f6f614555cc58d4e14b03d832572325df6095b/tests/test_spherical_painting_methods.py#L314

File: pyproject.toml [hunk 1]
```diff
@@ -11,8 +11,12 @@ readme = "README.md"
 requires-python = ">=3.9"
 license = { file = "LICENSE" }
 urls = { "Homepage" = "https://github.com/DifferentiableUniverseInitiative/JaxPM" }
-dependencies = ["jax_cosmo", "jax>=0.4.35", "jaxdecomp>=0.2.3"]
-
+dependencies = [
+  "jax_cosmo",
+  "jax>=0.4.35",
+  "jaxdecomp>=0.2.8",
+  "jax-healpy @ git+https://github.com/CMBSciPol/jax-healpy@main"
```
[the remark this answers is not in the document]
Comment (ASKabalan) [reply pyproject.toml:18]: Very true

I requested a tag https://github.com/CMBSciPol/jax-healpy/issues/14#issuecomment-3315406722

File: tests/test_spherical_painting_methods.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+        "paint_nside": PAINT_NSIDE,
+        "udgrade_power": 0.0
+    }),
+    ("bilinear", {
+        "paint_nside": PAINT_NSIDE,
+        "udgrade_power": 0.0
+    }),
+    ("RBF_NEIGHBOR", {
+        "paint_nside": PAINT_NSIDE,
+        "udgrade_power": 0.0
```
[the remark this answers is not in the document]
Comment (ASKabalan) [reply tests/test_spherical_painting_methods.py:72]: udgrade is healpix function 
Not a typo
https://healpix.sourceforge.io/html/idl_ud_grade.htm

File: jaxpm/spherical.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+    R_min, R_max : float
+        Minimum and maximum comoving distance range to include
+    box_size : float or array
+        Size of the simulation box in physical units
+    mesh_shape : tuple
+        Shape of the simulation mesh (nx, ny, nz)
+    weights : ndarray, optional
+        Particle weights (default: uniform weights)
+    method : str
+        Painting method: 'ngp', 'bilinear', or 'rbf_neighbor'
```
[the remark this answers is not in the document]
Comment (ASKabalan) [reply jaxpm/spherical.py:349]: This analysis is wrong.
The method is converted to upper case so code runs good
